src/Symfony/Component/Messenger/DataCollector/BusCollector.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\DataCollector;
+
+use Symfony\Component\Messenger\MessageBusInterface;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class BusCollector implements MessageBusInterface
+{
+    private $decoratedBus;
+    private $collector;
+
+    public function __construct(MessageBusInterface $decoratedBus, MessengerDataCollector $collector)
```

Comment:
why not a traceable bus implementation instead and collect from the data collector, how it's usually done AFAIK :)